microservice-transaction-sample/order-service/src/main/java/sample/order/OrderService.java

```diff
@@ -221,80 +180,108 @@ private void callRollbackEndpoint(String transactionId) {
   /** Get Order information by order ID */
   @Override
   public void getOrder(GetOrderRequest request, StreamObserver<GetOrderResponse> responseObserver) {
-    DistributedTransaction transaction = null;
-    try {
-      // Start a transaction
-      transaction = transactionManager.start();
-
-      // Retrieve the order info for the specified order ID
-      Optional<Order> order = Order.getById(transaction, request.getOrderId());
-      if (!order.isPresent()) {
-        throw Status.NOT_FOUND.withDescription("Order not found").asRuntimeException();
-      }
+    execInTwoPhaseCommit(responseObserver, "Getting an order",
+        (transaction) -> {
+          // Retrieve the order info for the specified order ID
+          Optional<Order> order = Order.getById(transaction, request.getOrderId());
+          if (!order.isPresent()) {
+            throw Status.NOT_FOUND.withDescription("Order not found").asRuntimeException();
+          }
 
-      // Make an order protobuf to return
-      sample.rpc.Order rpcOrder = getOrder(transaction, order.get());
+          // Get the customer name from Customer service
+          String customerName = getCustomerName(transaction.getId(), order.get().customerId);
```

Comment:
This is to avoid calling `getCustomerInfoForTwoPhaseCommit()` multiple times which only joins a global transaction not resumes it. If `join()` is called multiple times, it fails due to `The transaction already exists`.